### DISCUSSION TITLE
Open inline hyperlinked URLs in new tab (markdown)

### DISCUSSION THREAD

**Discussion by gargajit** [fastn-stack/fastn#929] [Ideas & RFCs]
We should allow users to open the hyperlinked URL in a new tab to access the information of the reference URL and can continue their journey on the source page.

Currently, the URL loads on the same page.

**Comment by amitu** [comment]
@gargajit try the syntax options in https://stackoverflow.com/questions/4425198/can-i-create-links-with-target-blank-in-markdown to see if any works with us.

**Reply by gargajit** [reply]
Solutions that I got from the URL: 

**1. `Kramdown` support this:**   (tried, does not work)

```
[link](url){:target="_blank"}
```
Result:

![image](https://github.com/fastn-stack/fastn/assets/118595104/8efaee65-0678-42a3-bdf8-1e9ae0eb0a28)

**2. Another suggestion:**    (tried, does not work)

```
[Open Graph protocol](https://ogp.me/?target=_blank)
```
This also does not open in new tab.


### **_Other Possible Solutions_**

**3. Inject <base target="_blank"> in Header**  

An alternative solution would be to inject <base target="_blank"> in the HTML head section using the includes option:

- Create a new HTML file, name it for example links.html
```
<base target="_blank">
```

- Then update your _output.yml
```
bookdown::gitbook:
  includes:
    in_header: links.html
 ```  
    
**4. Through JS**

```
Array.from(javascript.links)
    .filter(link => link.hostname != window.location.hostname)
    .forEach(link => link.target = '_blank');
 ```
    
 **5. For React + Markdown environment:**

The guy has created a reusable component:
```
export type TargetBlankLinkProps = {
  label?: string;
  href?: string;
};

export const TargetBlankLink = ({
  label = "",
  href = "",
}: TargetBlankLinkProps) => (
  <a href={href} target="__blank">
    {label}
  </a>
);
```
